readers/common/src/main/java/com/criteo/hadoop/garmadon/reader/GarmadonReader.java

```diff
@@ -258,6 +259,70 @@ int get() {
         }
     }
 
+    public interface GarmadonConsumer<K, V> {
+        ConsumerRecords<K, V> poll(long timeout);
+
+        void commitSync(Map<TopicPartition, OffsetAndMetadata> offsets);
+
+        void commitAsync(Map<TopicPartition, OffsetAndMetadata> offsets, OffsetCommitCallback callback);
+    }
+
+    private static final class UnsafeConsumer<K, V> implements GarmadonConsumer<K, V> {
+
+        private final Consumer<K, V> consumer;
+
+        private UnsafeConsumer(Consumer<K, V> consumer) {
+            this.consumer = consumer;
+        }
+
+        public ConsumerRecords<K, V> poll(long timeout) {
+            return consumer.poll(timeout);
+        }
+
+        public void commitSync(Map<TopicPartition, OffsetAndMetadata> offsets) {
+            consumer.commitSync(offsets);
+        }
+
+        public void commitAsync(Map<TopicPartition, OffsetAndMetadata> offsets, OffsetCommitCallback callback) {
+            consumer.commitAsync(offsets, callback);
+        }
+
+        static <K, V> UnsafeConsumer<K, V> unsafe(Consumer<K, V> consumer) {
+            return new UnsafeConsumer<>(consumer);
+        }
+    }
+
+    private static final class SynchronizedConsumer<K, V> implements GarmadonConsumer<K, V> {
+
+        private final Consumer<K, V> consumer;
+
+        private SynchronizedConsumer(Consumer<K, V> consumer) {
+            this.consumer = consumer;
+        }
+
+        public synchronized ConsumerRecords<K, V> poll(long timeout) {
+            synchronized (consumer) {
```

Comment:
Why the double synchronization?

Reply:
it was simple copy paste of how the code was before.
I removed the duplicated synchronized.
Thanks for the review.

Reply:
For the record, it was because in HdfsReader we were synchronizing on the consumer instance. 
Allowing to create a consumer outside of GarmadoReader scope was necessary to access other methods of the consumer before starting the read process (subscribe with listeners). 
SynchronizedConsumer is supposed to be used only inside GarmadonReader to provide CommittableOffset facility. Since ReaderBuilder now accepts externally created consumer it is better to synchronize on the instance instead of the class like you did.

Reply:
It does not justify the double synchronize but I guess the person who did the commit was not confident enough ad did not want to break an existing behavior.